README.md

```diff
@@ -224,6 +224,10 @@ This extension includes the following presets:
   field
 * `"resource_format_autocomplete"` - resource format validation with
   format guessing based on url and autocompleting form field
+* `"json"` - JSON based input. Only JSON objects and arrays are supported.
```

Comment:
arrays are supported?

Reply:
Arrays are supported, if passed as a JSON string (eg `a_json_field='[{"a", 1}, {"b", 2}]'`), but not if passed directly (eg `a_json_field=[{'a', 1}, {'b', 2}]`) because of how navl handles the validation.

This is controlled at the [validator level](https://github.com/ckan/ckanext-scheming/pull/163/files#diff-f65a08a9d3340b20658060def4af6419R243) though, so it can be easy restricted if you think that might be an issue.

Reply:
the asymmetry is a problem: that would mean that you can't call package_patch on a dataset with a list in a json field because package_patch passes the data from package_show (returns a list) to package_update.

1. How about naming this field json_object instead of json and only accept and produce json objects? I can't think of a case where I'd want a metadata field to be either a list or an object.

2. lists can work with simple validators (not navl nested dict ones) I use them with the multiple select fields, ckanext-repeating and ckanext-composite does whole sub-forms with them. Again this might make more sense as a separate json_list preset that only accepts and produces lists. 

If we do support both lists and in one preset I can see someone arguing that we should support the supporting other valid JSON values (`true`, `false`, `null`, floats and strings) but that's ambiguous when also accepting json as a string, so I'd rather not.

Reply:
Sounds good, `json_object` it is.